server/src/main/java/org/opensearch/identity/ExtensionTokenProcessor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.identity;
+
+import java.security.Principal;
+
+/**
+ * Token processor class to handle token encryption/decryption
+ */
+public class ExtensionTokenProcessor {
+    private String extensionUniqueId;
+
+    public ExtensionTokenProcessor(String extensionUniqueId) {
+        this.extensionUniqueId = extensionUniqueId;
+    }
+
+    public String getExtensionUniqueId() {
+        return extensionUniqueId;
+    }
+
+    /**
+     * ENCRYPTION
+     *
+     * Create a two-way encrypted access token for given principal for this extension
+     * @return token generated from principal
+     */
+    public PrincipalIdentifierToken generateToken(Principal principal) {
+        String token = principal.getName() + ":" + extensionUniqueId;
```

Comment:
The javadoc says it's encrypted but this result isn't.  Add a comment (and Issue matching the comment) to add encryption later.   

Should we be generating a JWT here?

Reply:
I'm not sure about that yet, but we can cover that as part of adding encryption/decryption task.
I will update the javadoc